src/drivers/amd/renoir/acp_dmic_dai.c

```diff
@@ -51,6 +53,19 @@ static inline int acp_dmic_dai_set_config(struct dai *dai, struct ipc_config_dai
 		return -EINVAL;
 	}
 	io_reg_write(PU_REGISTER_BASE + ACP_WOV_CLK_CTRL, clk_ctrl.u32all);
+	switch (acpdata->dmic_params.pdm_ch) {
+	case 2:
+		pdm_channels.bits.pdm_no_of_channels = 0;
+		break;
+	case 4:
+		pdm_channels.bits.pdm_no_of_channels = 1;
+		break;
+	default:
+		dai_info(dai, "ACP:acp_dmic_set_config unsupported channels");
+		return -EINVAL;
```

Comment:
just to confirm - this is intentional, that in some cases invalid channel number causes an error and in others a default value is just used?

Reply:
changes done